lib/ts/recipe/dashboard/api/userdetails/userMetadataPut.ts

```diff
@@ -0,0 +1,48 @@
+import { APIInterface, APIOptions } from "../../types";
+import UserMetadaRecipe from "../../../usermetadata/recipe";
+import UserMetaData from "../../../usermetadata";
+import STError from "../../../../error";
+
+type Response = {
+    status: "OK";
+};
+
+export const userMetadataPut = async (_: APIInterface, options: APIOptions): Promise<Response> => {
+    const requestBody = await options.req.getJSONBody();
+    const userId = requestBody.userId;
+    const data = requestBody.data;
+
+    // This is to throw an error early in case the recipe has not been initialised
+    UserMetadaRecipe.getInstanceOrThrowError();
+
+    if (userId === undefined || typeof userId !== "string") {
+        throw new STError({
+            message: "Required parameter 'userId' is missing or has an invalid type",
+            type: STError.BAD_INPUT_ERROR,
+        });
+    }
+
+    if (data === undefined || typeof data !== "string") {
+        throw new STError({
+            message: "Required parameter 'data' is missing or has an invalid type",
+            type: STError.BAD_INPUT_ERROR,
+        });
+    }
+
+    // Make sure that data is a valid JSON, this will throw
+    try {
+        JSON.parse(data);
```

Comment:
if data is of type string, this will not throw.. you want to make sure data is of type JSON object

Reply:
`data` is from the request body and we want it to be a string. `JSON.parse("")` would throw an error since the string is not a valid JSON string